bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java

```diff
@@ -144,6 +144,7 @@ private TransactionLocalContext createTrxContextBranch(TransactionLocalContext c
                         currentTrxContext.getInfoRecord());
         String currentTrxBlockId = currentTrxContext.getCurrentTransactionBlockId();
         trxCtx.addCurrentTransactionBlockId(currentTrxBlockId + "_" + strandName);
+        trxCtx.setTransactionContextStore(currentTrxContext.getTransactionContextStore());
```

Comment:
This method is not calling from all strand constructors. Will that be a problem?

Reply:
That would be fine. There are two Strand constructs [1]9https://github.com/ballerina-platform/ballerina-lang/blob/cfde8eaf1657f0969eea220d4c6d042b391f0b99/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java#L94) and [2](https://github.com/ballerina-platform/ballerina-lang/blob/cfde8eaf1657f0969eea220d4c6d042b391f0b99/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java#L118).
Only [2] creates a new TransactionLocalContext. Other strand just propagates the property bag